server/src/main/java/org/opensearch/index/RemoteRefreshSegmentPressureTracker.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.common.util.MovingAverage;
+import org.opensearch.common.util.Streak;
+import org.opensearch.common.util.concurrent.ConcurrentCollections;
+import org.opensearch.index.shard.ShardId;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * Keeps track of remote refresh which happens in {@link org.opensearch.index.shard.RemoteStoreRefreshListener}. This consist of multiple critical metrics.
+ */
+public class RemoteRefreshSegmentPressureTracker {
```

Comment:
We can ignore `Pressure` from the name and simply name it `RemoteSegmentTracker`

Reply:
the constructor of this class accepts the pressure settings, i will refactor the constructor to accept just the moving average window size and make the change.